Explorer/Assets/Scripts/CrdtEcsBridge/Engine/SimpleFetchApiImplementation.cs

```diff
@@ -0,0 +1,121 @@
+﻿using Arch.Core;
+using CommunicationData.URLHelpers;
+using Cysharp.Threading.Tasks;
+using DCL.Diagnostics;
+using DCL.WebRequests;
+using Microsoft.ClearScript;
+using Microsoft.ClearScript.JavaScript;
+using Nethereum.Contracts;
+using SceneRuntime.Apis;
+using SceneRuntime.Apis.Modules;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using UnityEngine;
+using UnityEngine.Networking;
+
+namespace CrdtEcsBridge.Engine
+{
+    public class SimpleFetchApiImplementation : ISimpleFetchApi
+    {
+        private enum RequestMethod
+        {
+            GET,
+            POST,
+            PUT,
+            PATCH,
+            HEAD,
+            INVALID,
+        }
+
+        private readonly Dictionary<string, string> headersDictionary = new ();
+
+        public void Dispose() { }
+
+        public async UniTask<ISimpleFetchApi.FetchResponse> Fetch(string requestMethod, string url, object headers, bool hasBody, string body,
+            string redirect, int timeout, IWebRequestController webController, CancellationToken ct)
+        {
+            RequestMethod parsedRequestMethod = ParseRequestMethod(requestMethod);
+
+            if (parsedRequestMethod == RequestMethod.INVALID) { throw new ArgumentException("Invalid request method."); }
+
+            var commonArguments = new CommonArguments(URLAddress.FromString(url), null, 3, timeout);
+            var webRequestHeaders = new WebRequestHeadersInfo();
+            headersDictionary.Clear();
+
+            if (headers is IScriptObject scriptObject)
+            {
+                IEnumerable<string> propertyNames = scriptObject.PropertyNames;
+
+                foreach (string name in propertyNames)
+                {
+                    var property = scriptObject.GetProperty(name).ToString();
+                    webRequestHeaders.Add(name, property);
+                    headersDictionary.Add(name, property);
+                }
+            }
+
+            ITypedWebRequest request;
+
+            switch (parsedRequestMethod)
+            {
+                case RequestMethod.GET:
+                    request = await webController.GetAsync(commonArguments, ct, ReportCategory.SCENE_FETCH_REQUEST, webRequestHeaders);
+                    GenerateResponse(request);
```

Comment:
This way `request` in every branch will be boxed to cast to `ITypedWebRequest`. Change `GenerateResponse` to work with a generic constraint parameter to avoid this.